python/paddle/fluid/layers/nn.py

```diff
@@ -387,12 +395,19 @@ def dynamic_lstm(input,
     cell = helper.create_tmp_variable(dtype)
     batch_gate = helper.create_tmp_variable(dtype)
     batch_cell_pre_act = helper.create_tmp_variable(dtype)
+    inputs = {'Input': input, 'Weight': weight, 'Bias': bias}
+    if h_0:
+        assert h_0.shape == (-1, size), \
```

Comment:
The first dimension represents placeholder of batch size and may not be -1,  `input.shape[0]` might be better than `-1` thus we needn't assume users' batch size setting.

Reply:
Yeah, if the shape has been checked in the `InferShape`, maybe the Python doesn't need to check it.

Reply:
`Infershape` doesn't check the shape, we can check it in Python code, it's more clear for the user.